operator/pkg/ciliumendpointslice/endpointslice.go

```diff
@@ -313,6 +314,65 @@ func (c *SlimController) onIdentityDelete(cid *cilium_api_v2.CiliumIdentity) {
 	c.enqueueCESReconciliation(touchedCESs)
 }
 
+// On Pod Update, verify all the necessary fields are set.
+// We recalculate the relevant fields when updating the CES instead of
+// saving them here in case of any changes in value, to minimize the
+// number of CES updates.
+// Returns error if requires retry without pod update.
+func (c *SlimController) onPodUpdate(pod *slim_corev1.Pod) error {
+	if pod.GetName() == "" || pod.GetNamespace() == "" {
+		return nil
+	}
+
+	if pod.Spec.HostNetwork { // no CEP for host networking pods
+		return nil
+	}
+
+	_, err := GetPodEndpointNetworking(pod)
+	if err != nil {
+		c.logger.Debug("could not get endpointnetworking for pod",
+			logfields.K8sPodName, pod.Name,
+			logfields.Error, err)
+		// When pod is assigned IPs or scheduled, we will receive a new update.
+		return nil
+	}
+
+	node, err := c.reconciler.getNodeNameForPod(pod)
+	if err != nil {
+		c.logger.Debug("could not get node name for pod",
+			logfields.K8sPodName, pod.Name,
+			logfields.Error, err)
+		// When pod is scheduled, we will receive a new update.
+		return nil
+	}
+
+	cidKey, err := c.reconciler.getPodCIDKey(pod)
+	if err != nil {
+		c.logger.Debug("could not get labels for pod",
+			logfields.K8sPodName, pod.Name,
+			logfields.Error, err)
+		return err
+	}
+
+	pCid, err := c.reconciler.getPodIdentity(cidKey)
+	if err != nil {
+		// Pod CID couldn't be retrieved yet. We store known pod information
+		// in the ces cache, so it can be associated with the CID once it is
+		// created.
+		c.manager.AddPodMapping(pod, node, cidKey)
+		return err
```

Comment:
Is it correct/necessary to return an error? I'd expect reconciliation to be triggered when the identity appears, and not require a retry here